### PR TITLE
Add simple namer plugin

### PR DESCRIPTION
**Pull request by padmaia**
Hooked up a default namer plugin. Kept it simple since it will need to change when the output of the bundler changes.

### CODE REVIEW

File: packages/core/core/src/Parcel.js [hunk 1]
```diff
@@ -212,8 +212,12 @@ export default class Parcel {
   }
 
   // TODO: implement bundle types
-  package(bundles: any[]) {
-    return Promise.all(bundles.map(bundle => this.runPackage(bundle)));
+  package(bundleMap: any[]) {
```
Comment (fathyb) [inline packages/core/core/src/Parcel.js]: `any[]` -> `any`

File: packages/core/core/src/BundlerRunner.js [hunk 2]
```diff
@@ -8,7 +8,22 @@ export default class BundlerRunner {
...
+
+    for (let bundle of bundles) {
+      let name;
+      for (let namer of namers) {
+        name = namer.name(bundle);
+        if (name) break;
+      }
+
+      if (!name) throw new Error('Unable to name bundle');
+      bundleMap[name] = bundle;
```
[the remark this answers is not in the document]
Comment (padmaia) [reply packages/core/core/src/BundlerRunner.js]: My thinking was that `bundleMap` would turn into `bundleGraph` and I will come back and refactor this when that happens.

[the remark this answers is not in the document]
Comment (devongovett) [reply packages/core/core/src/BundlerRunner.js]: Wouldn't the bundle graph need to be created in the Bundler plugin, not the namer?

[the remark this answers is not in the document]
Comment (devongovett) [reply packages/core/core/src/BundlerRunner.js]: Also seems like regardless of where the bundle graph is made, having the filename of the bundle as part of the bundle object might be pretty useful.

[the remark this answers is not in the document]
Comment (padmaia) [reply packages/core/core/src/BundlerRunner.js]: > Wouldn't the bundle graph need to be created in the Bundler plugin, not the namer?

Yeah, my point was just that we are planning on changing how the bundles are represented soon, so we can worry about how the name gets attached when we make that change. The key to this PR was just to wire things up. This is where the naming implementation will live, but the implementation will change, so don't get too tied up in those details right now. Your suggestion to keep the name directly on the bundles is noted though and I will keep it in mind.

**Comment by mischnic** [comment]
Superseded by #2424

**Pull request closed without merging**